packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -81,6 +83,43 @@ export class ProblemWidget extends TreeWidget {
         return 'caption';
     }
 
+    protected renderTailDecorations(node: TreeNode, props: NodeProps) {
+        return <div className='row-button-container'>
+            {this.renderRemoveButton(node)}
+        </div>;
+    }
+
+    protected renderRemoveButton(node: TreeNode): React.ReactNode {
+        return <span className='remove-node' onClick={e => this.remove(node, e)}></span>;
+    }
+
+    protected readonly remove = (node: TreeNode, e: React.MouseEvent<HTMLElement>) => this.doRemove(node, e);
+    protected doRemove(node: TreeNode, e: React.MouseEvent<HTMLElement>) {
+        this.removeNode(node);
+        e.stopPropagation();
+    }
+
+    protected removeNode(node: TreeNode) {
+        if (MarkerInfoNode.is(node)) {
+            this.problemManager.removeMarkers({ uri: node.uri });
+        }
+        if (MarkerNode.is(node)) {
+            this.problemManager.removeMarkers({
+                uri: new URI(node.marker.uri),
+                owner: node.marker.owner,
```

Comment:
Quick question: ATM I think we have a single source / owner for error markers: the Language Server that handles a given source file. If we had other entities generating markers (e.g. user-triggered build), would the markers from different owners, for the same file, end-up grouped together in the Problems widget? If so, Would we be able to clean them all? i.e. would the data filter be able to match all entries?

Reply:
Anyways, regarding your [comment](https://github.com/theia-ide/theia/pull/2954#issuecomment-423945890) I am also not sure if this PR is relevant anymore.

Reply:
@akosyakov do you have an opinion about the current need for this feature? It works well to momentarily clear markers, but depending how and by whom those markers were created, we may not control if or when they will be re-created. e.g. for most LS, it seems that just editing a file will trigger all markers in that file to be re-sent to Theia, to be re-created and re-added to the Problems view.

If for some reason we still end-up with stale markers in the Problems view, then it would help to be able to clear them. The biggest cause for this used to be when restoring markers from browser storage, but for the wrong workspace or git branch, but we do save/restore markers, anymore.